File: backend/models/Order.js
const mongoose = require('mongoose');

const OrderSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  items: { type: [{ name: String, qty: Number, price: Number }], default: [] },
  total: { type: Number, default: 0 },
  status: { type: String, default: 'pending' }
}, { timestamps: true });

module.exports = mongoose.models.Order || mongoose.model('Order', OrderSchema);

File: backend-java/src/main/java/com/organicfruits/api/models/Order.java
package com.organicfruits.api.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "orders")
public class Order {
    @Id
    private String id;
    private String orderNumber;
    private String userId;
    private List<OrderItem> items;
    private Double total;
    private String status; // "pending", "completed", etc.
    private String paymentMethod; // "upi", "card", "cod"
    private String paymentStatus; // "pending", "paid", "failed"
    private String customerName;
    private String phone;
    private String address;
    private String notes;
    private LocalDateTime createdAt;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class OrderItem {
        private String id;
        private String name;
        private Double price;
        private Integer quantity;
    }
}
